vue-app/src/components/CopyButton.vue

```diff
@@ -0,0 +1,89 @@
+<template>
+  <tooltip
+    v-if="text"
+    :position="position || 'bottom'"
+    :content="isCopied ? 'Copied!' : `Copy${type && ` ${type}`}`"
+  >
+    <div
+      :class="`${myClass || 'default'} ${hasBorder && 'border'}`"
+      @click="copyToClipboard"
+    >
+      <img width="16px" src="@/assets/copy.svg" />
+    </div>
+  </tooltip>
+</template>
+
+<script lang="ts">
+import Vue from 'vue'
+import Component from 'vue-class-component'
+import { Prop } from 'vue-property-decorator'
+
+import Tooltip from '@/components/Tooltip.vue'
+
+@Component({ components: { Tooltip } })
+export default class CopyButton extends Vue {  
+  @Prop() text!: string // Required: Text to copy
+  @Prop() type!: string // Optional: Fills in "Copy ____" in tooltip
+  @Prop() position!: string // Optional: Position of tooltip (default "bottom")
+  @Prop() myClass!: string // Optional class override for custom styling
+  @Prop() hasBorder!: boolean
+  @Prop() callback!: (value: boolean) => void
+
+  isLoading = false
+  isCopied = false
+
+  async copyToClipboard (): Promise<void> {
+    this.isLoading = true
+    try {
+      await navigator.clipboard.writeText(this.text)
+      this.isLoading = false
+      this.isCopied = true
+      if (typeof this.callback !== 'undefined') {
+        this.callback(true)
```

Comment:
As you are not requiring this `callback` prop, I encourage you to take a look at https://vuejs.org/v2/guide/components-custom-events.html

You could do something like this:
```
this.$emit('copied', true || false)

<copy-button v-on:copied="handleCopied"></copy-button>

handleCopied(isCopied: boolean) {
  ...
}
```

If you prefer using a function prop, I would suggest having a more explicit name for the callback. Something like `onCopied`.